### ISSUE TITLE
Ansible requires ssh for localhost even with ansible_connection=local

### ISSUE THREAD

**Issue by palasso** [ansible/ansible#11695]
##### Issue Type:
- Bug Report
##### Ansible Version:

ansible 1.9.2
configured module search path = None
##### Ansible Configuration:

```
# /etc/ansible/hosts
localhost ansible_connection=local ansible_python_interpreter=/usr/bin/python2
```
##### Environment:

Archlinux (running ansible locally)
##### Summary:

Ansible can't run playbooks locally without ssh if `ansible_connection=local` is defined in the hosts file, although it can run playbooks locally without ssh with `connection: local` in the playbook or with flag `--connection=local`.
##### Steps To Reproduce:
- Create `test.yml`:

```

---
- name: Test ping on localhost

  hosts: localhost
  connection: local

  tasks:

  - name: Ping!
    ping:
```
- run: `ansible-playbook -v test.yml`
- Remove the line `connection: local` from `test.yml`
- run: `ansible-playbook -v test.yml --connection=local`

Notice that until now it works perfectly fine.
- Run `ansible-playbook -v test.yml` and notice output:

```
Traceback (most recent call last):
  File "/usr/bin/ansible", line 197, in <module>
    (runner, results) = cli.run(options, args)
  File "/usr/bin/ansible", line 163, in run
    extra_vars=extra_vars,
  File "/usr/lib/python2.7/site-packages/ansible/runner/__init__.py", line 233, in __init__
    cmd = subprocess.Popen(['ssh','-o','ControlPersist'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
```
- Install `ssh` (e.g. on archlinux: `pacman -S openssh`) and run again the command `ansible-playbook -v test.yml`. Notice it works again.
##### Expected Results:

Ansible being able to run locally without the need to have ssh installed.
##### Actual Results:

Ansible **requires ssh** when `ansible_connection=local` is defined in the hosts file.
Ansible **doesn't require ssh** for a playbook when run with flag `--connection=local`.
Ansible **doesn't require ssh** for a playbook when declared `connection: local` in the playbook.

**Comment by sheldonh** [comment]
Also present in devel:

```
ansible 2.0.0 (devel b09f1f8e0b) last updated 2015/07/22 11:16:29 (GMT +200)
  lib/ansible/modules/core: (detached HEAD 55bc9e8fb1) last updated 2015/07/22 11:16:32 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD 26c1bd76c3) last updated 2015/07/22 11:16:32 (GMT +200)
  v1/ansible/modules/core: (detached HEAD f8d8af17cd) last updated 2015/07/17 15:26:48 (GMT +200)
  v1/ansible/modules/extras: (detached HEAD 495ad450e5) last updated 2015/07/17 15:26:52 (GMT +200)
  configured module search path = None
```

**Comment by jimi-c** [comment]
# Closing This Ticket

Hi!

We believe the above commit should resolve this problem for you. This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by jimi-c** [comment]
Also merged into stable-1.9, so this will be included in 1.9.3.
